lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -388,11 +433,16 @@ def _list_backups(self) -> Dict[int, str]:
             for snapshot in response.get("snapshots", [])
         }
 
-    def is_backup_in_progress(self, backup_id=None) -> bool:
-        """Returns True if backup is in progress, False otherwise."""
-        return self._query_backup_status(backup_id) in [
-            BackupServiceState.SNAPSHOT_IN_PROGRESS,
-        ]
+    def is_backup_in_progress(self) -> bool:
+        """Returns True if backup is in progress, False otherwise.
+
+        We filter the _query_backup_status() and seek for the following states:
+        - SNAPSHOT_IN_PROGRESS
+        """
+        if self._query_backup_status() in [BackupServiceState.SNAPSHOT_IN_PROGRESS]:
+            # We have a backup in progress
+            return True
+        return False
 
     def _query_backup_status(self, backup_id=None) -> BackupServiceState:
```

Comment:
Please add a retry here - and catch the OpenSearchHttpError - and return the right state if such exception occurs

Reply:
The `_query_backup_status` is in charge of returning a state. This method returns a boolean. So, what I will do is apply your ask to the `_query_backup_status`